### PR TITLE
Refactor htgettoken script into a Python module with an entry point

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR refactors the `htgettoken` script into a Python module (importable as `import htgettoken`) with a command-line entry point built on the fly when the package is installed.

### CODE REVIEW

**Comment by DrDaveD** [comment]
Is this complete, ready to merge as far as you are concerned?  What is the library API?  Should there be a document describing that included?

I tried building an rpm, and it looks like it is not downloading anything from the internet during download.  Is that how you understand it as well?  That's important for koji build.

**Comment by duncanmmacleod** [comment]
> Is this complete, ready to merge as far as you are concerned? What is the library API? Should there be a document describing that included?

This is complete and ready to merge as far as I am concerned. There is no documented library API, but users should be able to run something like this:

```python
import htgettoken
htgettoken.main()
```

which should be identical to the command-line entry point. I was holding off to a future PR to suggest any actual changes to the code.

> 
> I tried building an rpm, and it looks like it is not downloading anything from the internet during download. Is that how you understand it as well? That's important for koji build.

Yes, that should be the case.

File: htgettoken.spec [hunk 1]
```diff
@@ -76,6 +78,7 @@ rm -rf $RPM_BUILD_ROOT
 #   That used to be an additional optional meaning of the --vaultalias option,
 #   but urllib3 requires only one name to match.
 # - Add setuptools build infrastructure
+# - Use wheels to build/install Python package
```
Comment (DrDaveD) [inline htgettoken.spec:81]: How about being a little more descriptive about the implications of this, something like `, resulting in making htgettoken.main` directly callable by python programs.

Comment (duncanmmacleod) [reply htgettoken.spec]: Sorry for the delay, this is now done. Building with wheels doesn't impact the way that the package can be used, but it does slightly improve the available metadata and means that entry point scripts aren't built to rely upon `pkg_resources`.

Comment (DrDaveD) [reply htgettoken.spec]: Oh, so it's the previous line "Add setuptools build infrastucture" that made it possible to invoke htgettoken.main from a python program?  If so, how about adding that fact onto that entry?

Comment (duncanmmacleod) [reply htgettoken.spec]: @DrDaveD, sorry for the delay.

> Oh, so it's the previous line "Add setuptools build infrastucture" that made it possible to invoke htgettoken.main from a python program? If so, how about adding that fact onto that entry?

Not quite, the `htgettoken.main` is independent of those (sort of*). I have now added an extra spec changelog entry to describe the refactoring itself, I hope this is satisfactory.

<sub>*Building with wheels only has useful implications when an entry point is created, so refactoring the script into a module doesn't require using wheels (or vice-versa) but using wheels is only useful after this refactoring.</sub>

**Review by DrDaveD — APPROVED** [review]

**Pull request merged**